### PR TITLE
Update npm tests to use mocha test framework

### PR DESCRIPTION
**Pull request by burtonr**
Signed-off-by: Burton Rheutan <rheutan7@gmail.com>

## Description

As requested, updated the tests for the npm package to use the [Mocha test framework](https://mochajs.org/)

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
This request came from @alexellis in a comment on issue #740 

## How Has This Been Tested?

Ran `npm test` and `npm run test:install` locally

- `npm test`
```
> @openfaas/faas-cli@0.11.3-2 test /home/burtonr/code/go/src/github.com/openfaas/faas-cli/npm
> mocha "./test.js"

  Platform Suffix
    ✓ should return .exe for Windows_NT
    ✓ should return blank for Linux x64
    ✓ should return -arm64 for Linux aarch64
    ✓ should return -armhf for Linux armv61
    ✓ should return -armhf for Linux armv71
    ✓ should return -darwin for MacOS
    ✓ should throw for unsupported platform

  Binary Name
    ✓ should return faas-cli.exe for Windows_NT
    ✓ should return faas-cli for Linux
    ✓ should return faas-cli for Darwin

  Release Response
    ✓ should get a result from release (427ms)


  11 passing (436ms)

```

- `npm run test:install`
```
> @openfaas/faas-cli@0.11.3-2 test:install /home/burtonr/code/go/src/github.com/openfaas/faas-cli/npm
> mocha "./test-install.js"

  E2E Install
Downloading package https://github.com/openfaas/faas-cli/releases/download/0.11.3/faas-cli to /home/burtonr/code/go/src/github.com/openfaas/faas-cli/npm/bin/faas-cli
Download complete.
    ✓ should install binary (4626ms)


  1 passing (5s)

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
@Waterdrips PTAL

File: npm/test-install.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const lib = require('./lib');
+const assert = require('assert');
+
+describe('E2E Install', function () {
+    it('should install binary', async function() {
+        this.timeout(5000);
+        await installer.install();
+    
+        let binaryName = lib.getBinaryName(os.type(), os.arch());
+        let dest = path.join(__dirname, `bin/${binaryName}`);
```
Comment (Waterdrips) [inline npm/test-install.js]: Should probably pass `bin` as it's own arg to path.join

`let dest = path.join(__dirname, 'bin', binaryName)`

Comment (burtonr) [reply npm/test-install.js]: Good catch. Updated

Comment (Waterdrips) [reply npm/test-install.js]: Looks like you changed the one in install.js but not npm/test-install.js

Comment (burtonr) [reply npm/test-install.js]: I didn't realize that same command was also in the test... updated there as well :+1:

**Comment by alexellis** [comment]
On Linux with or without sudo is failing for me

```
[alex@nuc ~]$ sudo npm i -g @openfaas/faas-cli
/usr/bin/faas -> /usr/lib/node_modules/@openfaas/faas-cli/cli.js
/usr/bin/faas-cli -> /usr/lib/node_modules/@openfaas/faas-cli/cli.js

> @openfaas/faas-cli@0.11.3-3 postinstall /usr/lib/node_modules/@openfaas/faas-cli
> node -e 'require("./install.js").init()'

EACCES: permission denied, mkdir '/usr/lib/node_modules/@openfaas/faas-cli/bin'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @openfaas/faas-cli@0.11.3-3 postinstall: `node -e 'require("./install.js").init()'`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @openfaas/faas-cli@0.11.3-3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-01-15T15_45_10_340Z-debug.log
```

**Comment by alexellis** [comment]
Also fails on MacOS

```
Last login: Wed Jan 15 13:05:50 on ttys001
space-mini:~ alex$ npm i -g @openfaas/faas-cli
npm WARN checkPermissions Missing write access to /usr/local/lib/node_modules
npm ERR! path /usr/local/lib/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/usr/local/lib/node_modules'
npm ERR!  { [Error: EACCES: permission denied, access '/usr/local/lib/node_modules']
npm ERR!   stack:
npm ERR!    'Error: EACCES: permission denied, access \'/usr/local/lib/node_modules\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/local/lib/node_modules' }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alex/.npm/_logs/2020-01-15T15_46_33_846Z-debug.log
space-mini:~ alex$ sudo npm i -g @openfaas/faas-cli
Password:
Sorry, try again.
Password:
/usr/local/bin/faas -> /usr/local/lib/node_modules/@openfaas/faas-cli/cli.js
/usr/local/bin/faas-cli -> /usr/local/lib/node_modules/@openfaas/faas-cli/cli.js

> @openfaas/faas-cli@0.11.3-3 postinstall /usr/local/lib/node_modules/@openfaas/faas-cli
> node -e 'require("./install.js").init()'

EACCES: permission denied, mkdir '/usr/local/lib/node_modules/@openfaas/faas-cli/bin'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @openfaas/faas-cli@0.11.3-3 postinstall: `node -e 'require("./install.js").init()'`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @openfaas/faas-cli@0.11.3-3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alex/.npm/_logs/2020-01-15T15_46_50_609Z-debug.log
space-mini:~ alex$ 
```

@Waterdrips @burtonr am I missing a step, how is your testing method different from mine?

**Pull request merged**